### PR TITLE
Simplify EPSG find and always handle antimerridian

### PR DESCRIPTION
**Pull request by jhkennedy**
* use `multimode` to determine hemisphere; in the event of a tie, choose lowest valued (North).
* re-implement median to choose lowest zone of median two when an even number of zones are given
* no numpy dep. for this now

### CODE REVIEW

File: asf_tools/composite.py [hunk 1]
```diff
@@ -40,19 +40,17 @@ def get_target_epsg_code(codes: List[int]) -> int:
...
-    if target_zone == 0:
-        target_zone = 60
+    hemispheres = [c // 100 * 100 for c in codes]
+    # if even modes, choose lowest (North)
+    target_hemisphere = min(multimode(hemispheres))
 
-    target_hemisphere = int(statistics.mode(hemispheres))
+    zones = [c % 100 for c in codes]
+    # if even modes, choose lowest
+    target_zone = min(multimode(zones))
```
Comment (jhkennedy) [inline asf_tools/composite.py:53]: `min(multimode(...))` ensures we're always getting a valid zone (sidestepping the antimeridian problem), but now, may do something else undesirable. E.g., 

```python
>>> zones = [32731, 32631, 32732, 32633, 32733]
>>> get_target_epsg_code(zones)
32731
```
Or worse:
```python
>>> zones = [32731, 32631, 32732, 32732, 32633, 32733]
>>> get_target_epsg_code(zones)
32731
```
I both of these, to minimize distortion we'd at `32732` (our first test case might be a little contrived, but even modes across many zones might be possible)

Comment (jhkennedy) [reply asf_tools/composite.py:53]: `median(multimode(...))` would give us the same problems we started with

File: asf_tools/composite.py [hunk 2]
```diff
@@ -40,19 +40,17 @@ def get_target_epsg_code(codes: List[int]) -> int:
     # UTM EPSG codes for each hemisphere will look like:
     #   North: 326XX
     #   South: 327XX
-    code_array = np.array(codes)
-    valid_codes = np.concatenate([np.arange(32601, 32661), np.arange(32701, 32761)])
-    if not np.isin(code_array, valid_codes).all():
+    valid_codes = list(range(32601, 32661)) + list(range(32701, 32761))
+    if not all([c in valid_codes for c in codes]):
```
Comment (asjohnston-asf) [inline asf_tools/composite.py:44]: consider:
```
if bad_codes := set(codes) - set(valid_codes):
    raise ValueError(f'Non UTM EPSG code(s) encountered: {bad_codes}')
```

**Review by asjohnston-asf — APPROVED** [review]

**Comment by asjohnston-asf** [comment]
Resolves https://github.com/ASFHyP3/asf-tools/issues/35

**Pull request merged**